XamarinCommunityToolkit/Actions/Snackbar/Snackbar.android.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using Xamarin.Forms;
+using Android.Widget;
+using Xamarin.Forms.Platform.Android;
+using Xamarin.CommunityToolkit.Extensions;
+#if MONOANDROID10_0
+using AppCompatAlertDialog = AndroidX.AppCompat.App.AlertDialog;
+using AppCompatActivity = AndroidX.AppCompat.App.AppCompatActivity;
+using AndroidSnackbar = Google.Android.Material.Snackbar.Snackbar;
+#else
+using AppCompatAlertDialog = global::Android.Support.V7.App.AlertDialog;
+using AppCompatActivity = global::Android.Support.V7.App.AppCompatActivity;
+using AndroidSnackbar = global::Android.Support.Design.Widget.Snackbar;
+#endif
+
+namespace Xamarin.CommunityToolkit.Actions.Snackbar
+{
+	using global::Android.App;
+
+	public class SnackBar
+	{
+		public SnackBar(Activity subscriber)
+		{
+			MessagingCenter.Subscribe<Page, SnackbarArguments>(subscriber, PageExtension.SnackbarSignalName, OnSnackbarRequested);
+		}
+
+		void OnSnackbarRequested(Page sender, SnackbarArguments arguments)
+		{
+			var view = Platform.GetRenderer(sender).View;
+			var snackbar = AndroidSnackbar.Make(view, arguments.Message, arguments.Duration);
+			var snackbarView = snackbar.View;
+			var snackTextView = snackbarView.FindViewById<TextView>(Resource.Id.snackbar_text);
+			snackTextView.SetMaxLines(10);
+
+			if (!string.IsNullOrEmpty(arguments.ActionButtonText) && arguments.Action != null)
+			{
+				snackbar.SetAction(arguments.ActionButtonText, async (v) => await arguments.Action());
+			}
+
+			snackbar.AddCallback(new SnackbarCallback(arguments));
+			snackbar.Show();
+		}
+	}
+
+	class SnackbarCallback : AndroidSnackbar.BaseCallback
```

Comment:
would be great to have an own file for each class.

Reply:
it is private class only for android. I prefer to keep it here.

Reply:
but seems it's an internal class